City-of-Helsinki/infraohjelmointi-ui
# HKISD-22: fix budget saving and schedule editing from timeline

- Ticket: [https://futurice.atlassian.net/browse/HKISD-22](https://futurice.atlassian.net/browse/HKISD-22)
## The problem
Project-update and finance-update events where setting the start year of the finance data for the project that was edited to the current year, even though not all user always have the start year of the timeline as current year. There were also some use effects in the frontend that set the startYear of projects to current year and not the actual start year the user has selected. The PlanningRow component also had it's own state containing project which sometimes didn't mach the global state and it caused problems.
## The solution
- added a converter to convert the finance data that the backend sends to start from the year that the user has selected and added the finance year to be send with the data when updating projects schedule
- removed the projects from PlanningRow's own state and started using the global state there as well
- edited the eslint rules to show warnings of missing dependencies from dependency arrays
## How to test
- move the timeline and edit project schedules and budget and check that it works correctly
- open another window and check that updates made on the other window are showing correctly in the other one (try setting timelines to start from different years)
- Check that after updating something and then moving the timeline, all projects are showing and not just the one that you modified
